gio/src/socket.rs

```diff
@@ -27,22 +29,24 @@ impl Socket {
             if error.is_null() {
                 Ok(from_glib_full(ret))
             } else {
-                libc::close(fd);
+                let _ = OwnedFd::from_raw_fd(fd);
```

Comment:
Ultimately, `Drop` on OwnedFd does libc::close, no strong opinion though.

https://doc.rust-lang.org/src/std/os/fd/owned.rs.html#182

Reply:
It does, yes. But it seems more symmetric to close it via `OwnedFd` if we got it from there :)